src/esp/gfx/DepthUnprojection.h

```diff
@@ -0,0 +1,141 @@
+     * passed in @ref setProjectionMatrix().
+     */
+    UnprojectExistingDepth = 1 << 0,
+
+    /**
+     * By default, depth values on the far plane are patched to have a value of
+     * @cpp 0.0f @ce (instead of @cpp 100.0f @ce or whatever the far plane is
+     * set to). This might have some performance penalty and can be turned off
+     * with this flag.
+     */
+    NoFarPlanePatching = 1 << 1
+  };
+
+  /** @brief Flags */
+  typedef Corrade::Containers::EnumSet<Flag> Flags;
+
+  /** @brief Constructor */
+  explicit DepthShader(Flags flags = {});
+
+  /**
+   * @brief Set transformation and projection matrix
+   * @return Reference to self (for method chaining)
+   *
+   * Expects that @ref Flag::UnprojectExistingDepth is not set.
+   */
+  DepthShader& setTransformationMatrix(const Magnum::Matrix4& matrix);
+
+  /**
+   * @brief Set projection matrix for unprojection
+   * @return Reference to self (for method chaining)
+   */
+  DepthShader& setProjectionMatrix(const Magnum::Matrix4& matrix);
+
+  /**
+   * @brief Bind depth texture
+   * @return Reference to self (for method chaining)
+   *
+   * Expects that @ref Flag::UnprojectExistingDepth is set.
+   */
+  DepthShader& bindDepthTexture(Magnum::GL::Texture2D& texture);
+
+ private:
+  Flags flags_;
+  int transformationMatrixUniform_, projectionMatrixOrDepthUnprojectionUniform_;
+};
+
+CORRADE_ENUMSET_OPERATORS(DepthShader::Flags)
+
+/**
+@brief Calculate depth unprojection coefficients for @ref unprojectDepth()
+
+Given a projection transformation of vector @f$ \boldsymbol{v} @f$ using a
+matrix @f$ \boldsymbol{P} @f$ where @f$ x @f$ and @f$ y @f$ are arbitrary, the
+transformation can be reduced to a 2x2 matrix multiplication: @f[
+  \begin{array}{rcl}
+    \boldsymbol{P} \boldsymbol{v} & = & \begin{pmatrix}
+      p & 0 & 0 & 0 \\
+      0 & q & 0 & 0 \\
+      0 & 0 & a & b \\
+      0 & 0 & -1 & 0
+    \end{pmatrix} \begin{pmatrix}
+      x \\ y \\ z \\ 1
+    \end{pmatrix} \\[2.5em]
+
+    \boldsymbol{P'} \boldsymbol{v} & = & \begin{pmatrix}
+      a & b \\
+      -1 & 0
+    \end{pmatrix} \begin{pmatrix}
+      z \\ 1
+    \end{pmatrix} = \begin{pmatrix}
+      az + b \\
+      -z
+    \end{pmatrix}
+  \end{array}
+@f]
+
+In the OpenGL depth buffer, the depth values @f$ d @f$ are in range
+@f$ [ 0 ; 1 ] @f$ and the final depth value is after a perspective divide. The
+output has Z going forward, not backward, so we're looking for the value of
+@f$ -z @f$: @f[
+  \begin{array}{rcl}
+    2d - 1 & = & \cfrac{az + b}{-z} \\[1.25em]
+    -z & = & \cfrac{b}{2d + a - 1}
+  \end{array}
+@f]
+
+Finally, to reduce the amount of operations in @ref unprojectDepth(), we
+integrate the constants into @f$ a @f$ and @f$ b @f$, returning @f$ a' @f$ and
+@f$ b' @f$: @f[
+  \begin{array}{rcl}
+    -z & = & \cfrac{b}{2d + (a - 1)} \\[1.25em]
+    -z & = & \cfrac{\frac{1}{2}b}{d + \frac{1}{2}(a - 1)} \\[1.25em]
+    -z & = & \cfrac{b'}{d - a'} ; ~~~~ a' = \frac{1}{2}(a - 1),
```

Comment:
d + a'

Reply:
Oh well. Thanks a lot. Making typos in important documentation and then preserving them *forever* in a screenshot is my speciality :see_no_evil: